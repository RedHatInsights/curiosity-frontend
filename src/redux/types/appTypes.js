/**
 * @memberof Types
 * @module AppTypes
 */

const STATUS_4XX = '4XX';
const STATUS_5XX = '5XX';
const SET_PRODUCT = 'SET_PRODUCT';
const SET_PRODUCT_VARIANT = 'SET_PRODUCT_VARIANT';
const SET_PRODUCT_VARIANT_QUERY_RESET_ALL = 'SET_PRODUCT_VARIANT_QUERY_RESET_ALL';

/**
 * Application action, reducer types.
 *
 * @type {{STATUS_4XX: string, SET_PRODUCT_VARIANT: string, SET_PRODUCT_VARIANT_QUERY_RESET_ALL: string,
 *     SET_PRODUCT: string, STATUS_5XX: string}}
 */
const appTypes = {
  STATUS_4XX,
  STATUS_5XX,
  SET_PRODUCT,
  SET_PRODUCT_VARIANT,
  SET_PRODUCT_VARIANT_QUERY_RESET_ALL
};

export {
  appTypes as default,
  appTypes,
  STATUS_4XX,
  STATUS_5XX,
  SET_PRODUCT,
  SET_PRODUCT_VARIANT,
  SET_PRODUCT_VARIANT_QUERY_RESET_ALL
};
