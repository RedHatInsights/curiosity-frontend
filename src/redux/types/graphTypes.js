/**
 * @memberof Types
 * @module GraphTypes
 */

const SET_GRAPH_LEGEND = 'SET_GRAPH_LEGEND';

/**
 * Graph action, reducer types.
 *
 * @type {{SET_GRAPH_LEGEND: string}}
 */
const graphTypes = {
  SET_GRAPH_LEGEND
};

export { graphTypes as default, graphTypes, SET_GRAPH_LEGEND };
