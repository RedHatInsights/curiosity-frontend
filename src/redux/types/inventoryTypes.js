const SET_INVENTORY_TAB = 'SET_INVENTORY_TAB';

/**
 * Inventory action, reducer types.
 *
 * @type {{SET_INVENTORY_TAB: string}}
 */
const inventoryTypes = {
  SET_INVENTORY_TAB
};

export { inventoryTypes as default, inventoryTypes, SET_INVENTORY_TAB };
