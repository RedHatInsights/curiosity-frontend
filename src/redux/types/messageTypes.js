/**
 * @memberof Types
 * @module MessageTypes
 */

const SET_BANNER_MESSAGES = 'SET_BANNER_MESSAGES';

/**
 * User action, reducer types.
 *
 * @type {{SET_BANNER_MESSAGES: string}}
 */
const messageTypes = { SET_BANNER_MESSAGES };

export { messageTypes as default, messageTypes, SET_BANNER_MESSAGES };
