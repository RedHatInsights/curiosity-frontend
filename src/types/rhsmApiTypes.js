/**
 * RHSM response Capacity DATA type.
 *
 * @type {string}
 */
const RHSM_API_RESPONSE_CAPACITY_DATA = 'data';

/**
 * RHSM response Capacity DATA types.
 * Schema/map of expected Capacity data response properties.
 *
 * @type {{HYPERVISOR_SOCKETS: string, CORES: string, DATE: string, SOCKETS: string, PHYSICAL_SOCKETS: string,
 *     HYPERVISOR_CORES: string, HAS_INFINITE: string, PHYSICAL_CORES: string}}
 */
const RHSM_API_RESPONSE_CAPACITY_DATA_TYPES = {
  CLOUD_CORES: 'cloud_cores',
  CLOUD_INSTANCES: 'cloud_instance_count',
  CLOUD_SOCKETS: 'cloud_sockets',
  CORES: 'cores',
  DATE: 'date',
  HYPERVISOR_CORES: 'hypervisor_cores',
  HYPERVISOR_SOCKETS: 'hypervisor_sockets',
  PHYSICAL_CORES: 'physical_cores',
  PHYSICAL_SOCKETS: 'physical_sockets',
  SOCKETS: 'sockets',
  HAS_INFINITE: 'has_infinite_quantity'
};

/**
 * RHSM response Capacity META type.
 *
 * @type {string}
 */
const RHSM_API_RESPONSE_CAPACITY_META = 'meta';

/**
 * RHSM response Capacity META types.
 * Schema/map of expected Capacity META response properties.
 *
 * @type {{COUNT: string}}
 */
const RHSM_API_RESPONSE_CAPACITY_META_TYPES = {
  COUNT: 'count'
};

/**
 * RHSM response Error DATA type.
 *
 * @type {string}
 */
const RHSM_API_RESPONSE_ERROR_DATA = 'errors';

/**
 * RHSM response Error DATA types.
 * Schema/map of expected Error data response properties.
 *
 * @type {{CODE: string, DETAIL: string}}
 */
const RHSM_API_RESPONSE_ERROR_DATA_TYPES = {
  CODE: 'code',
  DETAIL: 'detail'
};

/**
 * RHSM response Error DATA CODE types.
 *
 * @type {{GENERIC: string, OPTIN: string}}
 */
const RHSM_API_RESPONSE_ERROR_DATA_CODE_TYPES = {
  GENERIC: 'SUBSCRIPTIONS1003',
  OPTIN: 'SUBSCRIPTIONS1004'
};

/**
 * RHSM response inventory DATA type.
 *
 * @type {string}
 */
const RHSM_API_RESPONSE_INVENTORY_DATA = 'data';

/**
 * RHSM response inventory DATA types.
 * Schema/map of expected inventory DATA response properties.
 *
 * @type {{CORES: string, HARDWARE: string, SOCKETS: string, ID: string, NAME: string, LAST_SEEN: string}}
 */
const RHSM_API_RESPONSE_INVENTORY_DATA_TYPES = {
  ID: 'insights_id',
  NAME: 'display_name',
  CORES: 'cores',
  SOCKETS: 'sockets',
  HARDWARE: 'hardware_type',
  LAST_SEEN: 'last_seen'
};

/**
 * RHSM response inventory type.
 *
 * @type {string}
 */
const RHSM_API_RESPONSE_INVENTORY_META = 'meta';

/**
 * RHSM response inventory META types.
 * Schema/map of expected Reporting/Tally META response properties.
 *
 * @type {{COUNT: string}}
 */
const RHSM_API_RESPONSE_INVENTORY_META_TYPES = {
  COUNT: 'count'
};

/**
 * RHSM response Reporting/Tally DATA type.
 *
 * @type {string}
 */
const RHSM_API_RESPONSE_PRODUCTS_DATA = 'data';

/**
 * RHSM response Reporting/Tally DATA types.
 * Schema/map of expected Reporting/Tally DATA response properties.
 *
 * @type {{HYPERVISOR_SOCKETS: string, CORES: string, DATE: string, SOCKETS: string, HAS_DATA: string,
 *     PHYSICAL_SOCKETS: string, HYPERVISOR_CORES: string, PHYSICAL_CORES: string}}
 */
const RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES = {
  CLOUD_CORES: 'cloud_cores',
  CLOUD_INSTANCES: 'cloud_instance_count',
  CLOUD_SOCKETS: 'cloud_sockets',
  CORES: 'cores',
  DATE: 'date',
  HYPERVISOR_CORES: 'hypervisor_cores',
  HYPERVISOR_SOCKETS: 'hypervisor_sockets',
  PHYSICAL_CORES: 'physical_cores',
  PHYSICAL_SOCKETS: 'physical_sockets',
  SOCKETS: 'sockets',
  HAS_DATA: 'has_data'
};

/**
 * RHSM response Reporting/Tally META type.
 *
 * @type {string}
 */
const RHSM_API_RESPONSE_PRODUCTS_META = 'meta';

/**
 * RHSM response Reporting/Tally META types.
 * Schema/map of expected Reporting/Tally META response properties.
 *
 * @type {{COUNT: string}}
 */
const RHSM_API_RESPONSE_PRODUCTS_META_TYPES = {
  COUNT: 'count'
};

/**
 * RHSM product id type values.
 *
 * @type {{RHEL_ARM: string, RHEL_WORKSTATION: string, RHEL_DESKTOP: string, RHEL: string,
 *     RHEL_SERVER: string, RHEL_IBM_Z: string, RHEL_COMPUTE_NODE: string, RHEL_IBM_POWER: string,
 *     RHEL_X86: string, OPENSHIFT: string}}
 */
const RHSM_API_PATH_ID_TYPES = {
  RHEL: 'RHEL',
  RHEL_COMPUTE_NODE: 'RHEL Compute Node',
  RHEL_DESKTOP: 'RHEL Desktop',
  RHEL_SERVER: 'RHEL Server',
  RHEL_WORKSTATION: 'RHEL Workstation',
  RHEL_ARM: 'RHEL for ARM',
  RHEL_IBM_POWER: 'RHEL for IBM Power',
  RHEL_IBM_Z: 'RHEL for IBM z',
  RHEL_X86: 'RHEL for x86',
  OPENSHIFT: 'OpenShift Container Platform'
};

/**
 * RHSM API query/search parameter GRANULARITY type.
 *
 * @type {string}
 */
const RHSM_API_QUERY_GRANULARITY = 'granularity';

/**
 * RHSM API query/search parameter of GRANULARITY type values.
 * Schema/map of expected query/search parameter granularity types.
 *
 * @type {{WEEKLY: string, QUARTERLY: string, DAILY: string, MONTHLY: string}}
 */
const RHSM_API_QUERY_GRANULARITY_TYPES = {
  DAILY: 'daily',
  WEEKLY: 'weekly',
  MONTHLY: 'monthly',
  QUARTERLY: 'quarterly'
};

/**
 * RHSM API query/search parameter results LIMIT type.
 *
 * @type {string}
 */
const RHSM_API_QUERY_LIMIT = 'limit';

/**
 * RHSM API query/search parameter result set OFFSET type.
 *
 * @type {string}
 */
const RHSM_API_QUERY_OFFSET = 'offset';

/**
 * RHSM API query/search parameter OPT-IN TALLY SYNC type.
 *
 * @type {string}
 */
const RHSM_API_QUERY_OPTIN_TALLY_SYNC = 'enable_tally_sync';

/**
 * RHSM API query/search parameter OPT-IN TALLY REPORT type.
 *
 * @type {string}
 */
const RHSM_API_QUERY_OPTIN_TALLY_REPORT = 'enable_tally_reporting';

/**
 * RHSM API query/search parameter OPTIN CONDUIT SYNC type.
 *
 * @type {string}
 */
const RHSM_API_QUERY_OPTIN_CONDUIT_SYNC = 'enable_conduit_sync';

/**
 * RHSM API query/search parameter SLA type.
 *
 * @type {string}
 */
const RHSM_API_QUERY_SLA = 'sla';

/**
 * RHSM API query/search parameter SLA type values.
 *
 * @type {{PREMIUM: string, SELF: string, NONE: string, STANDARD: string}}
 */
const RHSM_API_QUERY_SLA_TYPES = {
  PREMIUM: 'premium',
  STANDARD: 'standard',
  SELF: 'self-support',
  NONE: ''
};

/**
 * RHSM API query/search parameter START DATE type.
 * Associated with a sequential date prior/before the target date.
 *
 * @type {string}
 */
const RHSM_API_QUERY_START_DATE = 'beginning';

/**
 * RHSM API query/search parameter END DATE type.
 * Associated with a sequential date on or after the target date. Limited by the current date.
 *
 * @type {string}
 */
const RHSM_API_QUERY_END_DATE = 'ending';

/**
 * RHSM API types.
 *
 * @type {{RHSM_API_QUERY_END_DATE: string, RHSM_API_QUERY_START_DATE: string,
 *     RHSM_API_RESPONSE_ERROR_DATA_CODE_TYPES: {GENERIC: string, OPTIN: string},
 *     RHSM_API_RESPONSE_INVENTORY_DATA: string, RHSM_API_RESPONSE_CAPACITY_DATA: string,
 *     RHSM_API_RESPONSE_ERROR_DATA_TYPES: {CODE: string, DETAIL: string},
 *     RHSM_API_QUERY_OPTIN_TALLY_REPORT: string, RHSM_API_QUERY_OPTIN_TALLY_SYNC: string,
 *     RHSM_API_RESPONSE_CAPACITY_META_TYPES: {COUNT: string},
 *     RHSM_API_RESPONSE_CAPACITY_DATA_TYPES: {HYPERVISOR_SOCKETS: string, CORES: string, DATE: string,
 *     SOCKETS: string, PHYSICAL_SOCKETS: string, HYPERVISOR_CORES: string, HAS_INFINITE: string,
 *     PHYSICAL_CORES: string}, RHSM_API_QUERY_LIMIT: string,
 *     RHSM_API_QUERY_GRANULARITY_TYPES: {WEEKLY: string, QUARTERLY: string, DAILY: string, MONTHLY: string},
 *     RHSM_API_RESPONSE_PRODUCTS_DATA: string, RHSM_API_QUERY_GRANULARITY: string,
 *     RHSM_API_RESPONSE_INVENTORY_META: string, RHSM_API_PATH_ID_TYPES: {RHEL_ARM: string,
 *     RHEL_WORKSTATION: string, RHEL_DESKTOP: string, RHEL: string, RHEL_SERVER: string, RHEL_IBM_Z: string,
 *     RHEL_COMPUTE_NODE: string, RHEL_IBM_POWER: string, RHEL_X86: string, OPENSHIFT: string},
 *     RHSM_API_QUERY_SLA: string, RHSM_API_QUERY_OPTIN_CONDUIT_SYNC: string,
 *     RHSM_API_QUERY_SLA_TYPES: {PREMIUM: string, SELF: string, NONE: string, STANDARD: string},
 *     RHSM_API_RESPONSE_CAPACITY_META: string,
 *     RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES: {HYPERVISOR_SOCKETS: string, CORES: string, DATE: string,
 *     SOCKETS: string, HAS_DATA: string, PHYSICAL_SOCKETS: string, HYPERVISOR_CORES: string,
 *     PHYSICAL_CORES: string}, RHSM_API_RESPONSE_PRODUCTS_META: string, RHSM_API_RESPONSE_ERROR_DATA: string,
 *     RHSM_API_RESPONSE_INVENTORY_META_TYPES: {COUNT: string},
 *     RHSM_API_RESPONSE_PRODUCTS_META_TYPES: {COUNT: string},
 *     RHSM_API_RESPONSE_INVENTORY_DATA_TYPES: {CORES: string, HARDWARE: string, SOCKETS: string, ID: string,
 *     NAME: string, LAST_SEEN: string}, RHSM_API_QUERY_OFFSET: string}}
 */
const rhsmApiTypes = {
  RHSM_API_RESPONSE_CAPACITY_DATA,
  RHSM_API_RESPONSE_CAPACITY_DATA_TYPES,
  RHSM_API_RESPONSE_CAPACITY_META,
  RHSM_API_RESPONSE_CAPACITY_META_TYPES,
  RHSM_API_RESPONSE_ERROR_DATA,
  RHSM_API_RESPONSE_ERROR_DATA_TYPES,
  RHSM_API_RESPONSE_ERROR_DATA_CODE_TYPES,
  RHSM_API_RESPONSE_INVENTORY_DATA,
  RHSM_API_RESPONSE_INVENTORY_DATA_TYPES,
  RHSM_API_RESPONSE_INVENTORY_META,
  RHSM_API_RESPONSE_INVENTORY_META_TYPES,
  RHSM_API_RESPONSE_PRODUCTS_DATA,
  RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES,
  RHSM_API_RESPONSE_PRODUCTS_META,
  RHSM_API_RESPONSE_PRODUCTS_META_TYPES,
  RHSM_API_PATH_ID_TYPES,
  RHSM_API_QUERY_GRANULARITY,
  RHSM_API_QUERY_GRANULARITY_TYPES,
  RHSM_API_QUERY_LIMIT,
  RHSM_API_QUERY_OFFSET,
  RHSM_API_QUERY_OPTIN_TALLY_SYNC,
  RHSM_API_QUERY_OPTIN_TALLY_REPORT,
  RHSM_API_QUERY_OPTIN_CONDUIT_SYNC,
  RHSM_API_QUERY_SLA,
  RHSM_API_QUERY_SLA_TYPES,
  RHSM_API_QUERY_START_DATE,
  RHSM_API_QUERY_END_DATE
};

export {
  rhsmApiTypes as default,
  rhsmApiTypes,
  RHSM_API_RESPONSE_CAPACITY_DATA,
  RHSM_API_RESPONSE_CAPACITY_DATA_TYPES,
  RHSM_API_RESPONSE_CAPACITY_META,
  RHSM_API_RESPONSE_CAPACITY_META_TYPES,
  RHSM_API_RESPONSE_ERROR_DATA,
  RHSM_API_RESPONSE_ERROR_DATA_TYPES,
  RHSM_API_RESPONSE_ERROR_DATA_CODE_TYPES,
  RHSM_API_RESPONSE_INVENTORY_DATA,
  RHSM_API_RESPONSE_INVENTORY_DATA_TYPES,
  RHSM_API_RESPONSE_INVENTORY_META,
  RHSM_API_RESPONSE_INVENTORY_META_TYPES,
  RHSM_API_RESPONSE_PRODUCTS_DATA,
  RHSM_API_RESPONSE_PRODUCTS_DATA_TYPES,
  RHSM_API_RESPONSE_PRODUCTS_META,
  RHSM_API_RESPONSE_PRODUCTS_META_TYPES,
  RHSM_API_PATH_ID_TYPES,
  RHSM_API_QUERY_GRANULARITY,
  RHSM_API_QUERY_GRANULARITY_TYPES,
  RHSM_API_QUERY_LIMIT,
  RHSM_API_QUERY_OFFSET,
  RHSM_API_QUERY_OPTIN_TALLY_SYNC,
  RHSM_API_QUERY_OPTIN_TALLY_REPORT,
  RHSM_API_QUERY_OPTIN_CONDUIT_SYNC,
  RHSM_API_QUERY_SLA,
  RHSM_API_QUERY_SLA_TYPES,
  RHSM_API_QUERY_START_DATE,
  RHSM_API_QUERY_END_DATE
};
