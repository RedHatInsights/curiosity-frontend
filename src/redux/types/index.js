import { appTypes } from './appTypes';
import { graphTypes } from './graphTypes';
import { platformTypes } from './platformTypes';
import { queryTypes } from './queryTypes';
import { rhsmTypes } from './rhsmTypes';
import { toolbarTypes } from './toolbarTypes';
import { userTypes } from './userTypes';

const reduxTypes = {
  app: appTypes,
  graph: graphTypes,
  platform: platformTypes,
  query: queryTypes,
  rhsm: rhsmTypes,
  toolbar: toolbarTypes,
  user: userTypes
};

export {
  reduxTypes as default,
  reduxTypes,
  appTypes,
  graphTypes,
  platformTypes,
  queryTypes,
  rhsmTypes,
  toolbarTypes,
  userTypes
};
