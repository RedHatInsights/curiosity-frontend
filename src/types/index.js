import { rhsmApiTypes } from './rhsmApiTypes';

const apiTypes = {
  rhsmApi: rhsmApiTypes
};

export { apiTypes as default, apiTypes, rhsmApiTypes };
