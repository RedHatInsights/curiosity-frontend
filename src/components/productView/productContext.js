import React from 'react';
import { useSelector } from '../../redux';
import { RHSM_API_QUERY_TYPES } from '../../types/rhsmApiTypes';

/**
 * Product context.
 *
 * @type {React.Context<{}>}
 */
const ProductContext = React.createContext({});

/**
 * Generate queries.
 *
 * @param {string} queryType
 * @returns {object}
 */
const useQueryContext = (queryType = 'query') => {
  const { [queryType]: initialQuery, productId, viewId } = React.useContext(ProductContext) || {};
  const queryProduct = useSelector(({ view }) => view?.[queryType]?.[productId]);
  const queryView = useSelector(({ view }) => view?.[queryType]?.[viewId]);

  return {
    ...initialQuery,
    ...queryProduct,
    ...queryView
  };
};

/**
 * Expose query.
 *
 * @returns {object}
 */
const useQuery = () => useQueryContext();

/**
 * Expose query used for tally/capacity graph.
 *
 * @returns {object}
 */
const useGraphTallyQuery = () => ({ ...useQuery(), ...useQueryContext('graphTallyQuery') });

/**
 * Expose query used for hosts inventory.
 *
 * @returns {object}
 */
const useInventoryHostsQuery = () => ({ ...useQuery(), ...useQueryContext('inventoryHostsQuery') });

/**
 * Expose query used for subscriptions inventory.
 *
 * @returns {object}
 */
const useInventorySubscriptionsQuery = () => ({ ...useQuery(), ...useQueryContext('inventorySubscriptionsQuery') });

/**
 * Expose a filtered product context, blends context with Redux state layer via queries.
 *
 * @returns {object}
 */
const useProductContext = () => {
  const productConfig = React.useContext(ProductContext);
  const { [RHSM_API_QUERY_TYPES.UOM]: uomFilter } = useQuery();

  const {
    initialGraphFilters = [],
    initialInventoryFilters = [],
    initialSubscriptionsInventoryFilters = [],
    productContextFilterUom
  } = productConfig || {};

  if (productContextFilterUom) {
    const applyUomFilter = () => {
      const filterFilters = ({ id, isOptional }) => {
        if (!isOptional) {
          return true;
        }
        return new RegExp(uomFilter, 'i').test(id);
      };

      return {
        ...productConfig,
        initialGraphFilters: initialGraphFilters.filter(filterFilters),
        initialInventoryFilters: initialInventoryFilters.filter(filterFilters),
        initialSubscriptionsInventoryFilters: initialSubscriptionsInventoryFilters.filter(filterFilters)
      };
    };

    return applyUomFilter();
  }

  return productConfig;
};

export {
  ProductContext as default,
  ProductContext,
  useProductContext,
  useQueryContext,
  useQuery,
  useGraphTallyQuery,
  useInventoryHostsQuery,
  useInventorySubscriptionsQuery
};
