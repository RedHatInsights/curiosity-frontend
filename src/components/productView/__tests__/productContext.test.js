import React from 'react';
import { shallow } from 'enzyme';
import * as productContext from '../productContext';
import { RHSM_API_QUERY_TYPES, RHSM_API_QUERY_UOM_TYPES } from '../../../types/rhsmApiTypes';

describe('RouterContext', () => {
  // eslint-disable-next-line
  const Hook = ({ hook }) => <React.Fragment>{JSON.stringify(hook(), null, 2)}</React.Fragment>;
  const setContext = (context = {}) => jest.spyOn(React, 'useContext').mockImplementation(() => context);

  it('should return specific properties', () => {
    expect(productContext).toMatchSnapshot('specific properties');
  });

  it('should apply hooks for querying specific api endpoints', () => {
    const testContext = {
      productId: 'lorem',
      viewId: 'ipsum',
      query: { dolor: 'sit' },
      graphTallyQuery: { lorem: 'ipsum' },
      inventoryHostsQuery: { sit: 'amet' },
      inventorySubscriptionsQuery: { et: 'elementum' }
    };

    const generateHookResponse = ({ context, hook, message }) => {
      const spy = setContext(context);
      const component = shallow(<Hook hook={hook} />);
      expect(component).toMatchSnapshot(message);
      spy.mockClear();
    };

    generateHookResponse({ context: testContext, hook: productContext.useQueryContext, message: 'useQueryContext' });
    generateHookResponse({ context: testContext, hook: productContext.useQuery, message: 'useQuery' });
    generateHookResponse({
      context: testContext,
      hook: productContext.useGraphTallyQuery,
      message: 'useGraphTallyQuery'
    });
    generateHookResponse({
      context: testContext,
      hook: productContext.useInventoryHostsQuery,
      message: 'useInventoryHostsQuery'
    });
    generateHookResponse({
      context: testContext,
      hook: productContext.useInventorySubscriptionsQuery,
      message: 'useInventorySubscriptionsQuery'
    });
  });

  it('should apply a hook for useProductContext', () => {
    const testContext = {
      lorem: 'ipsum',
      productContextFilterUom: true,
      query: { [RHSM_API_QUERY_TYPES.UOM]: RHSM_API_QUERY_UOM_TYPES.CORES },
      initialGraphFilters: [
        { id: RHSM_API_QUERY_UOM_TYPES.CORES, isOptional: true, value: 'hello' },
        { id: RHSM_API_QUERY_UOM_TYPES.SOCKETS, isOptional: true, value: 'world' },
        { id: 'test id', value: 'test value' }
      ]
    };

    const generateHookResponse = ({ context, hook, message }) => {
      const spy = setContext(context);
      const component = shallow(<Hook hook={hook} />);
      expect(component).toMatchSnapshot(message);
      spy.mockClear();
    };

    generateHookResponse({
      context: testContext,
      hook: productContext.useProductContext,
      message: 'useProductContext'
    });
    generateHookResponse({
      context: testContext,
      hook: productContext.useProductContext,
      message: `useProductContext with uom filter, ${RHSM_API_QUERY_UOM_TYPES.CORES}`
    });
    generateHookResponse({
      context: { ...testContext, query: { [RHSM_API_QUERY_TYPES.UOM]: RHSM_API_QUERY_UOM_TYPES.SOCKETS } },
      hook: productContext.useProductContext,
      message: `useProductContext with uom filter, ${RHSM_API_QUERY_UOM_TYPES.SOCKETS}`
    });
  });
});
