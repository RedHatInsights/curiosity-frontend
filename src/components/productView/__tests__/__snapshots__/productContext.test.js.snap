// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouterContext should apply a hook for useProductContext: useProductContext 1`] = `
<Fragment>
  {
  "lorem": "ipsum",
  "productContextFilterUom": true,
  "query": {
    "uom": "cores"
  },
  "initialGraphFilters": [
    {
      "id": "cores",
      "isOptional": true,
      "value": "hello"
    },
    {
      "id": "test id",
      "value": "test value"
    }
  ],
  "initialInventoryFilters": [],
  "initialSubscriptionsInventoryFilters": []
}
</Fragment>
`;

exports[`RouterContext should apply a hook for useProductContext: useProductContext with uom filter, cores 1`] = `
<Fragment>
  {
  "lorem": "ipsum",
  "productContextFilterUom": true,
  "query": {
    "uom": "cores"
  },
  "initialGraphFilters": [
    {
      "id": "cores",
      "isOptional": true,
      "value": "hello"
    },
    {
      "id": "test id",
      "value": "test value"
    }
  ],
  "initialInventoryFilters": [],
  "initialSubscriptionsInventoryFilters": []
}
</Fragment>
`;

exports[`RouterContext should apply a hook for useProductContext: useProductContext with uom filter, sockets 1`] = `
<Fragment>
  {
  "lorem": "ipsum",
  "productContextFilterUom": true,
  "query": {
    "uom": "sockets"
  },
  "initialGraphFilters": [
    {
      "id": "sockets",
      "isOptional": true,
      "value": "world"
    },
    {
      "id": "test id",
      "value": "test value"
    }
  ],
  "initialInventoryFilters": [],
  "initialSubscriptionsInventoryFilters": []
}
</Fragment>
`;

exports[`RouterContext should apply hooks for querying specific api endpoints: useGraphTallyQuery 1`] = `
<Fragment>
  {
  "dolor": "sit",
  "lorem": "ipsum"
}
</Fragment>
`;

exports[`RouterContext should apply hooks for querying specific api endpoints: useInventoryHostsQuery 1`] = `
<Fragment>
  {
  "dolor": "sit",
  "sit": "amet"
}
</Fragment>
`;

exports[`RouterContext should apply hooks for querying specific api endpoints: useInventorySubscriptionsQuery 1`] = `
<Fragment>
  {
  "dolor": "sit",
  "et": "elementum"
}
</Fragment>
`;

exports[`RouterContext should apply hooks for querying specific api endpoints: useQuery 1`] = `
<Fragment>
  {
  "dolor": "sit"
}
</Fragment>
`;

exports[`RouterContext should apply hooks for querying specific api endpoints: useQueryContext 1`] = `
<Fragment>
  {
  "dolor": "sit"
}
</Fragment>
`;

exports[`RouterContext should return specific properties: specific properties 1`] = `
Object {
  "ProductContext": Object {
    "$$typeof": Symbol(react.context),
    "Consumer": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": [Circular],
    },
    "Provider": Object {
      "$$typeof": Symbol(react.provider),
      "_context": [Circular],
    },
    "_calculateChangedBits": null,
    "_currentRenderer": null,
    "_currentRenderer2": null,
    "_currentValue": Object {},
    "_currentValue2": Object {},
    "_threadCount": 0,
  },
  "default": Object {
    "$$typeof": Symbol(react.context),
    "Consumer": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": [Circular],
    },
    "Provider": Object {
      "$$typeof": Symbol(react.provider),
      "_context": [Circular],
    },
    "_calculateChangedBits": null,
    "_currentRenderer": null,
    "_currentRenderer2": null,
    "_currentValue": Object {},
    "_currentValue2": Object {},
    "_threadCount": 0,
  },
  "useGraphTallyQuery": [Function],
  "useInventoryHostsQuery": [Function],
  "useInventorySubscriptionsQuery": [Function],
  "useProductContext": [Function],
  "useQuery": [Function],
  "useQueryContext": [Function],
}
`;
